include/boost/math/special_functions/beta.hpp

```diff
@@ -31,14 +32,16 @@ namespace detail{
 // Implementation of Beta(a,b) using the Lanczos approximation:
 //
 template <class T, class Lanczos, class Policy>
-T beta_imp(T a, T b, const Lanczos&, const Policy& pol)
+BOOST_MATH_GPU_ENABLED T beta_imp(T a, T b, const Lanczos&, const Policy& pol)
 {
    BOOST_MATH_STD_USING  // for ADL of std names
 
+   #ifndef BOOST_MATH_NO_EXCEPTIONS
```

Comment:
Shouldn't we keep this block, even with no exceptions, and let the error handlers return NaN?  Otherwise for invalid arguments, pretty much anything could happen, including infinite loops or whatever...